gluten-core/src/main/scala/org/apache/spark/softaffinity/SoftAffinity.scala

```diff
@@ -44,6 +45,25 @@ abstract class Affinity(val manager: AffinityManager) extends LogLevelUtil with
     }
   }
 
+  def getFilePartitionLocations(filePartition: FilePartition): Array[String] = {
+    val filePaths = filePartition.files.map(_.filePath.toString)
+    val preferredLocations = filePartition.preferredLocations()
+    if (shouldUseSoftAffinity(filePaths, preferredLocations)) {
+      if (manager.detectDuplicateReading) {
```

Comment:
yes, soft affiity is enabled by a separate config different one from velox local cache configs. We add a new API for velox, so it should not impact CH cases. In this new API, if `detectDuplicateReading` is true, new logic would be called, otherwise the original logic is executed.